### PR TITLE
Identity certificate handling changes

### PR DESCRIPTION
**Pull request by rnicoll**
Introduce changes in certificate management ahead of enforcing correct certificate paths for all identities:

* Remove the existing certificate path composition utility, and replace it with direct API calls as it's down to  two lines of code now.
* Registering an identity now accepts the full PartyAndCertificate so it can be verified.
* Identity certificates can now sign certificates (for signing transaction key's certificates)
* Standardise the certificate validation logic in the identity service
* Correct several places where the certificate ordering in CertPath is incorrect; target certificate is first, root  last, not vice-versa.

### CODE REVIEW

File: core/src/main/kotlin/net/corda/core/serialization/DefaultKryoCustomizer.kt [hunk 1]
```diff
@@ -73,6 +73,7 @@ object DefaultKryoCustomizer {
 
             noReferencesWithin<WireTransaction>()
 
+            register(sun.security.ec.ECPublicKeyImpl::class.java, PublicKeySerializer)
```
Comment (kchalkias) [inline core/src/main/kotlin/net/corda/core/serialization/DefaultKryoCustomizer.kt]: why is this required? we used to only serialise BC keys, which are already registered

Comment (rnicoll) [inline core/src/main/kotlin/net/corda/core/serialization/DefaultKryoCustomizer.kt]: I believe it's because we're now sending certificate paths around, and they contain the TLS certificate.

**Comment by rnicoll** [comment]
Closing to try breaking this down into smaller parts
